p2p/net/session.go

```diff
@@ -118,6 +122,20 @@ func (n *NetworkSessionImpl) EncryptGuard() *sync.Mutex {
 	return &n.encGuard
 }
 
+// Sign signs that input message using the sessions hmac and returns the signature or an error
+func (n *NetworkSessionImpl) Sign(data []byte) ([]byte, error) {
+	hm := hmac.New(sha256.New, n.keyM)
+	return hm.Sum(data), nil
+}
+
+// VerifySignature verifies that the given signature is a valid signature for the given data
+func (n *NetworkSessionImpl) VerifySignature(data []byte, sign []byte) bool {
+	hm := hmac.New(sha256.New, n.keyM)
```

Comment:
What happens if data or sign are null or zero len?
also, You can use size to optimise the verification